### ISSUE TITLE
Question: Any chance of a separate library for models?

### ISSUE THREAD

**Issue by HenningRoigaard** [confluentinc/confluent-kafka-dotnet#1355]
Any chance of a separate library for models, e.g., Confluent.Kafka.Models? One that does not depend on librdkafka.redist, and contains all interfaces and datamodels?

We have built a framework on top of Confluent.Kafka, which we are in the process of using in a quite large monolithic .Net Framework product consisting of 1000+ projects. We use the old csproj format, requiring that all transitive dependencies be unfolded when referencing a package, and we won’t be changing project format anytime soon. Besides the usual annoying side effects of the old project format in relation to nuget, we also have an issue with props-file in our many legacy tools, which requires some work on our part.

So in short, we would very much like to be able to control the librdkafka.redist reference in one library, while allowing all others to work with the types in Confluent.Kafka

Yes, I my motivation for such a packet is very specific, but perhaps others could also benefit from it?

**Comment by mhowlett** [comment]
thanks for the summary. an unfortunate consequence of the native dependency. it's unlikely we'll prioritize addressing this unless we see a critical mass of other people hitting the problem - this is the first time it's come up. maybe there is a solution whereby you load the library dynamically at run time or have a separate process for kafka communication and do IPC from your main apps. Neither ideal, but a couple of ideas for work-arounds.
